### ISSUE TITLE
StudyJob v1alpha2 API version

### ISSUE THREAD

**Issue by johnugeorge** [kubeflow/katib#370]
There are couple of changes that are suggested. We can incorporate all these changes in the next API version.

 - [x]  Consistent field naming #351 , #371 
 - [x]  Extending Status fields to contain all conditions #344 
 - [x]  Extending Status fields to have detailed info #356 #122 
 - [x]  Renaming RequestNumber, RequestCount #161 
 - [x] Custom TemplateSpec for workers/metrics #349 
 - [x] StudyName field is unnecessary #380

Please add if I missed out any.

/cc @richardsliu @YujiOshima @hougangliu 

----------------

Implementation (last updated 5/13/2019):
- [x] Controllers implementation #441
- [x] Metrics collection #442
- [x] New UI  #443
- [x] Suggestion algorithms implementation #407 
- [x] DB operations https://github.com/kubeflow/katib/issues/510
- [x] Support custom resources https://github.com/kubeflow/katib/pull/512
- [x] Update kustomize packages #513 
- [x] Add e2e tests #514 
- [x] #551

**Comment by hougangliu** [comment]
@johnugeorge  please also add #349

**Comment by johnugeorge** [comment]
@hougangliu can you create an issue ?

**Comment by hougangliu** [comment]
@johnugeorge Done #371

**Comment by johnugeorge** [comment]
It would be better if  all other current bigger PRs are reviewed and merged before moving to v1alpha2.  Else, PRs need be modified for newer version too.  

PRs which should be reviewed and merged:

@YujiOshima  #362 #352 
@DeeperMind #339 #345 
@Akado2009  #328

**Comment by richardsliu** [comment]
How about #346 ? It might require some API changes too.

#341 is not API related but would require controller side changes, and would be good to add as well.

**Comment by YujiOshima** [comment]
https://github.com/kubeflow/katib/issues/291 is also a big change for the logic of controller.
It would be good to add.

**Comment by johnugeorge** [comment]
@YujiOshima @richardsliu i was listing out issues that need API changes. Except  #346(not finalized yet), other mentioned issues are already tagged with 0.5.0 which can be tracked separately.  Other than issues included in https://github.com/kubeflow/katib/issues/370#issuecomment-462779691,  we will get all new controller changes in v1alpha2.

**Comment by YujiOshima** [comment]
@johnugeorge OK, Thanks.


> It would be better if all other current bigger PRs are reviewed and merged before moving to v1alpha2. Else, PRs need be modified for newer version too.

Every PRs include changes of DB scheme are blocked by https://github.com/kubeflow/katib/issues/359 .
First of all, we should fix this. See https://github.com/kubeflow/katib/pull/360

**Comment by jlewi** [comment]
Now that #381 is merged what is the remaining work here?

**Comment by richardsliu** [comment]
I think we should keep this issue open until the implementation is done. In any case this should be moved to 0.6.

**Comment by gaocegege** [comment]
Hi, all

Could we do the jobs parallel? It seems that the process is blocked by #435 

BTW I am free these days and can work on some items.

**Comment by richardsliu** [comment]
There are a few PRs in flight:
- #435 
- #456 
- #457 
- #438

I think after #457 is merged we can split up the remainder of the work.

**Comment by jlewi** [comment]
@richardsliu @johnugeorge @gaocegege Any update on this issue?

**Comment by hougangliu** [comment]
@jlewi we're still working on UI #438 and suggestion services #435 (maybe a new PR will replace it to take over), plan to make them done by the end of this week.

**Comment by jlewi** [comment]
What are the remaining changes to be code complete for v1alpha2?
Is someone working on porting over the suggestion algorithms to the new API?

**Comment by johnugeorge** [comment]
@jlewi  Yes. I have started on it

**Comment by richardsliu** [comment]
This should be done now.
/close
